docs/index.md

```diff
@@ -44,7 +44,7 @@ import express from 'express';
 const app = express();
 
 app.use(express.json());
-app.post('/user', (req: express.Request, res: express.Response) => {
```

Comment:
You're wrong. RequestHandler type exported from @types/express will make the custom req.customProp type safe if the user has extended the Express type delcaration. So there's no need for the explicitly mentioning types there (it's redundant). If you don't believe me you can try extending express namespace and making req.customProp type safe and then check if it's on the req object or not.

Thanks for reviewing.

Reply:
You told it by yourself. In certain situations we can not rely on type inference because the user has to do further steps to make the code work, such as namespace extension...etc. By specifying those types we are saying to a new user (not an expert one), the `req` object is `express.Request` and that is the type you eventually have to extend.